pallets/ah-migrator/src/xcm_config.rs

```diff
@@ -16,32 +16,289 @@
+pub mod common {
+	use super::*;
+	parameter_types! {
+		pub const AssetHubParaId: ParaId = ParaId::new(system_parachain::ASSET_HUB_ID);
+		pub const DotLocation: Location = Location::parent();
+	}
+
+	pub struct FellowshipEntities;
+	impl Contains<Location> for FellowshipEntities {
+		fn contains(location: &Location) -> bool {
+			matches!(
+			location.unpack(),
+			(
+				1,
+				[
+					Parachain(system_parachain::COLLECTIVES_ID),
+					Plurality { id: BodyId::Technical, .. }
+				]
+			) | (
+				1,
+				[
+					Parachain(system_parachain::COLLECTIVES_ID),
+					PalletInstance(
+						collectives_polkadot_runtime_constants::FELLOWSHIP_SALARY_PALLET_INDEX
+					)
+				]
+			) | (
+				1,
+				[
+					Parachain(system_parachain::COLLECTIVES_ID),
+					PalletInstance(
+						collectives_polkadot_runtime_constants::FELLOWSHIP_TREASURY_PALLET_INDEX
+					)
+				]
+			)
+		)
+		}
+	}
+
+	pub struct AmbassadorEntities;
+	impl Contains<Location> for AmbassadorEntities {
+		fn contains(location: &Location) -> bool {
+			matches!(
+			location.unpack(),
+			(
+				1,
+				[
+					Parachain(system_parachain::COLLECTIVES_ID),
+					PalletInstance(
+						collectives_polkadot_runtime_constants::AMBASSADOR_SALARY_PALLET_INDEX
+					)
+				]
+			) | (
+				1,
+				[
+					Parachain(system_parachain::COLLECTIVES_ID),
+					PalletInstance(
+						collectives_polkadot_runtime_constants::AMBASSADOR_TREASURY_PALLET_INDEX
+					)
+				]
+			)
+		)
+		}
+	}
+
+	// Teleport filters are a special case because we might want to have finer control over which
+	// one to use at fine-grained stages of the migration.
+
+	/// Cases where a remote origin is accepted as trusted Teleporter for a given asset:
+	///
+	/// - DOT with the parent Relay Chain and sibling system parachains; and
+	/// - Sibling parachains' assets from where they originate (as `ForeignCreators`).
+	pub type TrustedTeleportersBeforeAfter = (
+		ConcreteAssetFromSystem<DotLocation>,
+		IsForeignConcreteAsset<FromSiblingParachain<AssetHubParaId>>,
+	);
+
+	/// During migration we only allow teleports of foreign assets (not DOT).
+	///
+	/// - Sibling parachains' assets from where they originate (as `ForeignCreators`).
+	pub type TrustedTeleportersDuring =
+		IsForeignConcreteAsset<FromSiblingParachain<AssetHubParaId>>;
 }
 
-/// Cases where a remote origin is accepted as trusted Teleporter for a given asset:
-///
-/// - DOT with the parent Relay Chain and sibling system parachains; and
-/// - Sibling parachains' assets from where they originate (as `ForeignCreators`).
-pub type TrustedTeleportersBeforeAfter = (
-	ConcreteAssetFromSystem<DotLocation>,
-	IsForeignConcreteAsset<FromSiblingParachain<AssetHubParaId>>,
+mod before {
+	use super::{
+		common::{AmbassadorEntities, AssetHubParaId, FellowshipEntities},
+		*,
+	};
+
+	parameter_types! {
+		pub RelayTreasuryLocation: Location =
+			(Parent, PalletInstance(polkadot_runtime_constants::TREASURY_PALLET_ID)).into();
```

Comment:
`polkadot_runtime_constants::TREASURY_PALLET_ID` is not network agnostic,
should be probably feature flagged `"ahm-polkadot"` / `"ahm-westend"` for
polkadot_runtime_constants::TREASURY_PALLET_ID;
kusama_runtime_constants::TREASURY_PALLET_ID;
westend_runtime_constants::TREASURY_PALLET_ID;